### PR TITLE
daemon/config: ignore UTF-8 BOM in config JSON

### PR DESCRIPTION
**Pull request by neersighted**
**- What I did**
Strip a BOM when present in the config JSON.

[RFC 8259] allows for JSON implementations to optionally ignore a BOM when it helps with interoperability; do so in Moby as Notepad (the only text editor available out of the box in many versions of Windows server) insists on writing UTF-8 with a BOM.

  [RFC 8259]: https://tools.ietf.org/html/rfc8259#section-8.1

**- How I did it**
`bytes.TrimPrefix`

**- How to verify it**
Unit tests.

**- Description for the changelog**
* Moby now accepts config JSON files with a UTF-8 byte order mark present.

**- A picture of a cute animal (not mandatory but encouraged)**
beardy

### CODE REVIEW

**Review by thaJeztah — APPROVED** [review]
LGTM, but left one suggestion

File: daemon/config/config.go [hunk 1]
```diff
@@ -412,6 +412,11 @@ func getConflictFreeConfiguration(configFile string, flags *pflag.FlagSet) (*Con
 
 	var config Config
```
[the remark this answers is not in the document]
Comment (neersighted) [reply daemon/config/config.go:413]: I'm not sure we can move it; the bytes are used in both the `flag != nil` path as well as the fast path, and we need `&config` present this high for the early return on an empty config. We could return a new `Config` on line 423, but that doesn't get is much as we again use config in the big `flags != nil` block on line 468.

Unless you mean a different movement that I'm not seeing at a glance?

[the remark this answers is not in the document]
Comment (thaJeztah) [reply daemon/config/config.go:413]: Yeah, was looking at moving it after the `bytes.Trimspace()` as it's now in the middle between some of the "cleaning up" code (reading bytes, trimming BOM, trimming whitespace);

```go
	b = bytes.TrimSpace(b)


 	var config Config
 	if len(b) == 0 {
```

**Review by tianon — APPROVED** [review]
Gross, but seems pretty reasonable to me :sweat_smile: :heart:

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**